db/db.go

```diff
@@ -229,24 +234,13 @@ func (d *DB) NewCollection(config CollectionConfig) (*Collection, error) {
 		}
 	}
 
-	for _, cfg := range config.Indexes {
-		// @todo: Should check to make sure this is a valid field path for this schema
-		if err := c.AddIndex(cfg.Path, cfg.Unique); err != nil {
-			return nil, err
-		}
-	}
-
-	indexBytes, err := json.Marshal(config.Indexes)
-	if err != nil {
-		return nil, err
-	}
-	indexKey := dsDBIndexes.ChildString(config.Name)
-	exists, err = d.datastore.Has(indexKey)
-	if err != nil {
+	if err := c.AddIndex(IndexConfig{Path: idFieldName, Unique: true}); err != nil {
 		return nil, err
 	}
-	if !exists {
-		if err := d.datastore.Put(indexKey, indexBytes); err != nil {
+
+	for _, cfg := range config.Indexes {
+		// @todo: Should check to make sure this is a valid field path for this schema
```

Comment:
Did you write this todo @carsonfarmer ? Should we make a GH issue?

Reply:
Yes I did, and yes we should: https://github.com/textileio/go-threads/issues/291